File: Algorithm_643_Maximum Average Subarray I.cpp
#include <vector>
#include <climits>
using namespace std;
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int len=nums.size();
        int sum=0;
        for (int i=0;i<k;i++) sum+=nums[i];
        int maxsum=sum;
        for (int i=1;i+k<=len;i++)
        {
            sum-=nums[i-1];
            sum+=nums[i+k-1];
            if (sum>maxsum) maxsum=sum;
        }
        return double(maxsum)/k;
    }
};

File: Algorithm_643_Maximum Average Subarray I.java
// 思路：记录长度为k的子数组的所有元素之和的最大值，存在maxSum变量中
// 要返回的结果就是maxSum / k，注意要先将其中一个转换为double类型

class Solution {
    public double findMaxAverage(int[] nums, int k) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int curSum = 0;
        for (int i = 0; i < k; i++) {
            curSum += nums[i];
        }
        int maxSum = curSum;
        for (int i = k; i < nums.length; i++) {
            curSum = curSum - nums[i - k] + nums[i];
            maxSum = Math.max(maxSum, curSum);
        }
        return (double) maxSum / k;
    }
}